cypress/integration/brewery/ScenarioSharing.js

```diff
@@ -1,40 +1,211 @@
+    Scenarios.getScenarioValidateButton().should('be.visible').should('not.be.disabled');
+    Scenarios.getScenarioRejectButton().should('be.visible').should('not.be.disabled');
+    Scenarios.getScenarioCreationButton().should('be.visible').should('not.be.disabled');
+    ScenarioParameters.getParametersEditButton().should('be.visible').should('not.be.disabled');
+    ScenarioParameters.getLaunchButton().should('be.visible').should('not.be.disabled');
+    RolesEdition.getShareScenarioButton().should('be.visible').should('not.be.disabled').click();
+    RolesEdition.getShareScenarioDialogAgentsSelect().should('be.visible').should('not.be.disabled');
+    RolesEdition.getShareScenarioDialogFirstCancelButton().click();
+    ScenarioManager.switchToScenarioManager();
+
+    ScenarioManager.writeInFilter(UNSHARED_SCENARIOS_LIST[0].name);
+    ScenarioManager.getDeleteScenarioButton().should('be.visible').should('not.be.disabled');
+    ScenarioManager.getScenarioEditableLabelInEditMode(UNSHARED_SCENARIOS_LIST[0].id).should('not.exist');
+    ScenarioManager.getScenarioEditableLabel(UNSHARED_SCENARIOS_LIST[0].id).click();
+    ScenarioManager.getScenarioEditableLabelInEditMode(UNSHARED_SCENARIOS_LIST[0].id).should('exist');
+  });
+
+  it('is shown a message error when unauthorized changes are tried in scenario sharing pop-up', () => {
+    // TODO: open share pop-up and check error messages are displayed when unauthorized changes are made:
+    RolesEdition.getShareScenarioButton().should('be.visible').should('not.be.disabled').click();
+    RolesEdition.getRoleEditorAgentName(USER_EXAMPLE.email).should('have.text', USER_EXAMPLE.email);
+    RolesEdition.getSelectWithAction(USER_EXAMPLE.email).first().click();
+    RolesEdition.getSelectActionName().click();
+    RolesEdition.getNoAdminErrorMessage().should('be.visible');
+    RolesEdition.getShareScenarioDialogSubmitButton().should('be.disabled');
   });
 
   it('can add & remove people, and change roles in scenario sharing pop-up', () => {
     // TODO: open share pop-up, make some changes (add/remove/change role + workspace default role), confirm & refresh
     // page to check changes have been made
     // For local run with stubbing, page refresh could be replaced by switching to another scenario and then switching
     // back
+    const scenario = UNSHARED_SCENARIOS_LIST[0];
+
+    const grantedPermissions = [PERMISSIONS.READ, PERMISSIONS.READ_SECURITY];
+    const notGrantedPermissions = [
+      PERMISSIONS.LAUNCH,
+      PERMISSIONS.WRITE,
+      PERMISSIONS.VALIDATE,
+      PERMISSIONS.WRITE_SECURITY,
+      PERMISSIONS.DELETE,
+    ];
+
+    Scenarios.selectScenario(scenario.name, scenario.id);
+    RolesEdition.getShareScenarioButton().click();
+    RolesEdition.getShareScenarioDialogTitle().should('have.text', 'Share ' + scenario.name);
+    RolesEdition.getSelectWithAction('Workspace').click();
+    RolesEdition.getSelectOption(ROLES.SCENARIO.EDITOR).click();
```

Comment:
Could we create a generic action to factorize the 2 steps ?
- open a dropdown list 
- select an option in the list